module_2/src/com/epam/decompositionusingmethods/NumbersAnalyzer.java

```diff
@@ -0,0 +1,40 @@
+package com.epam.decompositionusingmethods;
+
+public class NumbersAnalyzer {
+
+    public static int findGreatestCommonDivisor(int firstNumber, int secondNumber) {
+        int divisor = -1;
+        if (firstNumber > 0 && secondNumber > 0) {
+            divisor = Math.min(firstNumber, secondNumber);
+            int dividend = Math.max(firstNumber, secondNumber);
+            int remainder;
+            while (dividend % divisor != 0) {
+                remainder = dividend % divisor;
+                dividend = divisor;
+                divisor = remainder;
+            }
+        }
+        return divisor;
+    }
+
+    public static int findGreatestCommonDivisor(int firstNumber, int secondNumber, int thirdNumber, int fourthNumber) {
+        int divisor = -1;
+        if (firstNumber > 0 && secondNumber > 0 && thirdNumber > 0 && fourthNumber > 0) {
+            int[] numbers = new int[]{firstNumber, secondNumber, thirdNumber, fourthNumber};
+            divisor = findGreatestCommonDivisor(numbers[0], numbers[1]);
+            for (int i = 2; i < numbers.length; i++) {
```

Comment:
I don't see a big need to include to this array the firstNumber and the secondNumber
```suggestion
            divisor = findGreatestCommonDivisor(firstNumber, secondNumber);
            final int[] otherNumbers = new int[]{thirdNumber, fourthNumber};
            for (int i = 0; i < otherNumbers.length; i++) {
```